### PR TITLE
[SPARK-40587][CONNECT] Support SELECT * in an explicit way in connect proto

### PR DESCRIPTION
**Pull request by amaliujia**
### What changes were proposed in this pull request?

Support `SELECT *` in an explicit way by connect proto.

### Why are the changes needed?

Current proto uses empty project list for `SELECT *`. However, this is an implicit way that it is hard to differentiate `not set` and `set but empty` (the latter is invalid plan). For longer term proto compatibility, we should always use explicit fields for passing through information.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

UT

### CODE REVIEW

File: connect/src/main/protobuf/spark/connect/expressions.proto [hunk 1]
```diff
@@ -155,4 +156,7 @@ message Expression {
     string expression = 1;
   }
 
+  // represent * (e.g. SELECT *)
+  message UnresolvedStar {
+  }
```
Comment (amaliujia) [inline connect/src/main/protobuf/spark/connect/expressions.proto:161]: We need to expand this message later to support full fields of the star:
```
case class UnresolvedStar(target: Option[Seq[String]]) extends Star with Unevaluable {

```

which is will be used to support `Struct`.

Comment (amaliujia) [reply connect/src/main/protobuf/spark/connect/expressions.proto:161]: one example is `structDf.select(xxhash64($"a", $"record.*")`

**Comment by amaliujia** [comment]
@HyukjinKwon @cloud-fan @grundprinzip

File: connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala [hunk 2]
```diff
@@ -96,7 +96,9 @@ class SparkConnectPlanner(plan: proto.Relation, session: SparkSession) {
       rel: proto.Project,
       common: Option[proto.RelationCommon]): LogicalPlan = {
     val baseRel = transformRelation(rel.getInput)
-    val projection = if (rel.getExpressionsCount == 0) {
+    // TODO: support the target field for *.
+    val projection =
+      if (rel.getExpressionsCount == 1&& rel.getExpressions(0).hasUnresolvedStar) {
```
Comment (cloud-fan) [inline connect/src/main/scala/org/apache/spark/sql/connect/planner/SparkConnectPlanner.scala:101]:
```suggestion
      if (rel.getExpressionsCount == 1 && rel.getExpressions(0).hasUnresolvedStar) {
```

**Review by cloud-fan — APPROVED** [review]

File: connect/src/main/protobuf/spark/connect/expressions.proto [hunk 3]
```diff
@@ -155,4 +156,7 @@ message Expression {
     string expression = 1;
   }
 
+  // represent * (e.g. SELECT *)
```
Comment (grundprinzip) [inline connect/src/main/protobuf/spark/connect/expressions.proto:159]:
```suggestion
  // UnresolvedStar is used to expand all the fields of a relation or struct.
```

**Review by grundprinzip — APPROVED** [review]
just a nit on comment.

**Comment by amaliujia** [comment]
@cloud-fan friendly ping

**Comment by cloud-fan** [comment]
thanks, merging to master!

**Review by HyukjinKwon — COMMENTED** [review]
LGTM2

**Pull request closed without merging**